### PR TITLE
[Service Connector] support user managed identity and service principal

### PR DESCRIPTION
**Pull request by xfz11**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Service Connector

File: src/serviceconnector-passwordless/azext_serviceconnector_passwordless/_credential_free.py [hunk 1]
```diff
@@ -66,23 +72,50 @@ def enable_mi_for_db_linker(cmd, source_id, target_id, auth_info, client_type, c
...
+                        'displayName')
+                except CLIInternalError as e:
+                    if 'AADSTS530003' in e.error_msg:
+                        logger.warning(
+                            'Please ask your IT department for help to join this device to Azure Active Directory.')
+                    raise e
+            elif auth_info['auth_type'] == AUTHTYPES[AUTH_TYPE.UserIdentity]:
+                mi_client_id = auth_info.get('client_id')
+                mi_sub_id = auth_info.get('subscription_id')
+                umi_info = run_cli_cmd(
```
Comment (yungezz) [inline src/serviceconnector-passwordless/azext_serviceconnector_passwordless/_credential_free.py:99]: does this step need extra permission?

File: src/serviceconnector-passwordless/azext_serviceconnector_passwordless/_credential_free.py [hunk 2]
```diff
@@ -66,23 +72,50 @@ def enable_mi_for_db_linker(cmd, source_id, target_id, auth_info, client_type, c
...
+                target_handler.identity_client_id = mi_client_id
+                target_handler.identity_name = umi_info[0].get('name')
+            elif auth_info['auth_type'] == AUTHTYPES[AUTH_TYPE.ServicePrincipalSecret]:
+                sp_client_id = auth_info.get('client_id')
+                sp_object_id = auth_info.get('principal_id')
+                sp_info = run_cli_cmd(
+                    'az ad sp show --id {}'.format(sp_client_id))
+                if sp_info is None:
+                    raise Exception(
+                        "No Service Principal found for client id {}".format(sp_client_id))
```
Comment (yungezz) [inline src/serviceconnector-passwordless/azext_serviceconnector_passwordless/_credential_free.py:115]:
```suggestion
                        "Not found the service principal with client id {}".format(sp_client_id))
```

File: src/serviceconnector-passwordless/azext_serviceconnector_passwordless/_credential_free.py [hunk 3]
```diff
@@ -302,12 +369,14 @@ def create_aad_user_in_mysql(self, connection_kwargs, query_list):
...
-            raise AzureConnectionError("Fail to connect mysql. " + str(e)) from e
+            raise AzureConnectionError(
+                "Fail to connect mysql. " + str(e)) from e
         if cursor is not None:
             try:
                 cursor.close()
             except Exception as e:  # pylint: disable=broad-except
-                raise CLIInternalError("connection close failed." + str(e)) from e
+                raise CLIInternalError(
+                    "connection close failed." + str(e)) from e
```
Comment (yungezz) [inline src/serviceconnector-passwordless/azext_serviceconnector_passwordless/_credential_free.py:379]:
```suggestion
                    "Connection close failed." + str(e)) from e
```

File: src/serviceconnector-passwordless/azext_serviceconnector_passwordless/_credential_free.py (hunk 1 above)
Comment (xfz11) [reply src/serviceconnector-passwordless/azext_serviceconnector_passwordless/_credential_free.py:100]: Add try catch for permission missing

File: src/serviceconnector-passwordless/azext_serviceconnector_passwordless/_credential_free.py [hunk 4]
```diff
@@ -354,6 +431,7 @@ def set_user_admin(self, user_object_id, **kwargs):
         # pylint: disable=not-an-iterable
         admins = run_cli_cmd(
             'az sql server ad-admin list --ids {}'.format(self.target_id))
+        print(admins)
```
Comment (yungezz) [inline src/serviceconnector-passwordless/azext_serviceconnector_passwordless/_credential_free.py:434]: remove debug info

**Review by yungezz — APPROVED** [review]

**Comment by xfz11** [comment]
@kairu-ms Could you help merge the PR?

**Pull request merged**